### DISCUSSION TITLE
acmcsuf.com ♥ RSS

### DISCUSSION THREAD

**Discussion by EthanThatOneKid** [acmcsufoss/acmcsuf.com#269] [Newsletter]
# acmcsuf.com ♥ RSS

**tl;dr**: Check out our RSS feed on [`https://acmcsuf.com/feed.xml`](https://acmcsuf.com/feed.xml)! 😎

## What is RSS?

RSS is a specification built on top of XML which stands for [_Really Simple Syndication_](https://en.wikipedia.org/wiki/RSS). Computer people tend to use RSS to keep track of things on the Internet that they want to be notified about when anything new is published. This technology is useful for the syndication of things like blog posts, but can really be used for all types of digital publications imaginable.

## How do we leverage RSS?

Here on acmcsuf.com, our blog feature is an outlet for our community to publish computer science-related posts or in-depth updates/reflections about community events. In order to make sure people are notified as soon as a new post is published, we must be RSS-compatible. Lucky for us, RSS is really, really simple and only uses a single HTTP endpoint. You can find ours on [`https://acmcsuf.com/feed.xml`](https://acmcsuf.com/feed.xml).

## How can you use RSS in your daily life?

There are many ways people choose to indulge in RSS feeds. One way is to use a simple piece of software found online (e.g. desktop application, browser extension, etc.). A simple search in your favorite search engine should give you an up-to-date result.

To subscribe to an RSS feed, all you need is a single HTTP endpoint commonly ending with `.rss` or `.xml` since RSS is expressed as an [XML (Extensible Markup Language](https://en.wikipedia.org/wiki/XML)) document. Copy our HTTP endpoint below to subscribe within your RSS feed manager. 

```
https://acmcsuf.com/feed.xml
```

**Note**: Other standards exist such as [Atom](https://en.wikipedia.org/wiki/Atom_(standard)) and [JSON feed](https://jsonfeed.org/version/1) and are often used as alternatives to RSS.

### Choosing RSS feeds to follow

You're free to follow any publication that you like as long as you can find it online for free! For starters, you have [`https://acmcsuf.com/feed.xml`](https://acmcsuf.com/feed.xml). If you have a favorite Subreddit, you can subscribe to its RSS feed using the RSS feed URL pattern, `https://reddit.com/r/<your_subreddit>.rss` (example: [`https://reddit.com/r/csuf.rss`](https://reddit.com/r/csuf.rss)). Most blogs and major Internet platforms incorporate RSS into their sites, so it is up to you to keep an eye out for the [commonly orange] RSS logo! This symbol indicates the existence of an RSS feed. 

[![RSS logo](https://user-images.githubusercontent.com/31261035/150892071-3ec94857-aa5b-4011-9d60-bd569ce1b7c0.png)](https://en.wikipedia.org/wiki/RSS)

### Making your own RSS feed manager

If you are looking into learning more about RSS hands-on, you can actually interface with any RSS feed on the web as long as your favorite programming language has a trustworthy library for parsing RSS feeds (or at least XML documents).

**Note**: If your language of choice does not have such a library, you will just have to implement the entire [RSS specification](https://www.rssboard.org/rss-2-0-1) yourself, which is not particularly trivial, so I’d recommend finding a language that is capable to make things easier for yourself. 

For instance, we use [node-rss](https://github.com/dylang/node-rss#readme) to format the RSS document found on [`https://acmcsuf.com/feed.xml`](https://acmcsuf.com/feed.xml). Below are some libraries I found on GitHub that you can start with.

#### RSS v2.0.1 Compatible APIs

Language | Library
---|---
Node.js | [node-rss](https://github.com/dylang/node-rss#readme)
Deno | [x/rss](https://deno.land/x/rss)
Python | [feedparser](https://github.com/kurtmckee/feedparser#readme)
Go | [gofeed](https://github.com/mmcdole/gofeed#readme) 

## See also

- <https://en.wikipedia.org/wiki/RSS#History>
- <https://en.wikipedia.org/wiki/Web_syndication>
- Documentary about the prolific programmer, co-founder of Reddit [Aaron Swartz](https://en.wikipedia.org/wiki/Aaron_Swartz) who co-wrote the RSS specification: [The Internet's Own Boy: The Story of Aaron Swartz | full movie (2014)](https://www.youtube.com/watch?v=9vz06QO3UkQ)

## Last updated

- **2023-10-22**: Add Deno RSS library.
- **2022-11-05**: Rename `/blog.xml` to `/feed.xml`.

---

_🎉 posted in celebration of launch of [v2.0 release](https://github.com/EthanThatOneKid/acmcsuf.com/releases/tag/v2.0) ([v2.0 milestone](https://github.com/EthanThatOneKid/acmcsuf.com/milestone/1))_
🔗 _self link: <https://acmcsuf.com/rss>_
